src/measure_theory/integral/divergence_theorem.lean

```diff
@@ -123,6 +129,232 @@ begin
+theorem integral_eq_of_has_deriv_within_at_off_countable_of_le (f f' : ℝ → E)
+  {a b : ℝ} (hle : a ≤ b) {s : set ℝ} (hs : countable s)
+  (Hc : ∀ x ∈ s, continuous_within_at f (Icc a b) x)
+  (Hd : ∀ x ∈ Icc a b \ s, has_deriv_within_at f (f' x) (Icc a b) x)
+  (Hi : interval_integrable f' volume a b) :
+  ∫ x in a..b, f' x = f b - f a :=
+begin
+  set e : ℝ ≃L[ℝ] ℝ¹ := (continuous_linear_equiv.fun_unique (fin 1) ℝ ℝ).symm,
+  have e_symm : ∀ x, e.symm x = x 0 := λ x, rfl,
+  set F' : ℝ → ℝ →L[ℝ] E := λ x, smul_right (1 : ℝ →L[ℝ] ℝ) (f' x),
+  have hF' : ∀ x y, F' x y = y • f' x := λ x y, rfl,
+  calc ∫ x in a..b, f' x = ∫ x in Icc a b, f' x :
+    by simp only [interval_integral.integral_of_le hle, set_integral_congr_set_ae Ioc_ae_eq_Icc]
+  ... = ∑ i : fin 1, ((∫ x in Icc (e a ∘ i.succ_above) (e b ∘ i.succ_above),
+        f (e.symm $ i.insert_nth (e b i) x)) -
+      (∫ x in Icc (e a ∘ i.succ_above) (e b ∘ i.succ_above),
+        f (e.symm $ i.insert_nth (e a i) x))) :
+    begin
+      refine integral_divergence_of_has_fderiv_within_at_off_countable_of_equiv e _ _
+        (λ _, f) (λ _, F') s hs a b hle (λ x hx i, Hc x hx) (λ x hx i, Hd x hx) _ _ _,
+      { exact λ x y, (order_iso.fun_unique (fin 1) ℝ).symm.le_iff_le },
+      { exact (volume_preserving_fun_unique (fin 1) ℝ).symm },
+      { intro x, rw [fin.sum_univ_one, hF', e_symm, pi.single_eq_same, one_smul] },
+      { rw [interval_integrable_iff_integrable_Ioc_of_le hle] at Hi,
+        exact Hi.congr_set_ae Ioc_ae_eq_Icc.symm }
+    end
+  ... = f b - f a :
+    begin
+      simp only [fin.sum_univ_one, e_symm],
+      have : ∀ (c : ℝ), const (fin 0) c = is_empty_elim := λ c, subsingleton.elim _ _,
+      simp [this, volume_pi, measure.pi_of_empty (λ _ : fin 0, volume)]
+    end
+end
+
+/-- **Fundamental theorem of calculus, part 2**. This version assumes that `f` is differentiable off
+a countable set `s`, and is continuous at the points of `s`.
+
+See also `measure_theory.interval_integral.integral_eq_sub_of_has_deriv_right` for a version that
+only assumes right differentiability of `f`.
+-/
+theorem integral_eq_of_has_deriv_within_at_off_countable (f f' : ℝ → E) {a b : ℝ} {s : set ℝ}
+  (hs : countable s) (Hc : ∀ x ∈ s, continuous_within_at f [a, b] x)
+  (Hd : ∀ x ∈ [a, b] \ s, has_deriv_within_at f (f' x) [a, b] x)
+  (Hi : interval_integrable f' volume a b) :
+  ∫ x in a..b, f' x = f b - f a :=
+begin
+  cases le_total a b with hab hab,
+  { simp only [interval_of_le hab] at *,
+    exact integral_eq_of_has_deriv_within_at_off_countable_of_le f f' hab hs Hc Hd Hi },
+  { simp only [interval_of_ge hab] at *,
+    rw [interval_integral.integral_symm, neg_eq_iff_neg_eq, neg_sub, eq_comm],
+    exact integral_eq_of_has_deriv_within_at_off_countable_of_le f f' hab hs Hc Hd Hi.symm }
+end
+
+/-- **Divergence theorem** for functions on the plane. It is formulated in terms of two functions
+`f g : ℝ × ℝ → E` and an integral over `Icc a b`, where `a b : ℝ × ℝ`, `a ≤ b`.
```

Comment:
I think it is worth expanding the docstring to make it more self-contained for casual readers that would stumble on this without knowing the context. Something like
```suggestion
/-- **Divergence theorem** for functions on the plane along rectangles. It is formulated in terms of two functions
`f g : ℝ × ℝ → E` and an integral over `Icc a b = [a.1, b.1] ×[a.2, b.2]`, where `a b : ℝ × ℝ`, `a ≤ b`. When thinking of
`f` and `g` as the two coordinates of a single function `F : ℝ × ℝ → E × E` and when `E = ℝ`, this is the usual statement
that the integral of the divergence of `F` inside the rectangle equals the integral of the normal derivative of `F` along
the boundary.
```
The same comment applies to the next lemma.